scripts/test.py

```diff
@@ -394,104 +444,29 @@ def main() -> int:
-    if not args.no_clean:
-        return_code = run_subprocess(
-            cmd=["make", "-C", PROJECT_LOCATION, "clean"],
-            timeout=BUILD_TIMEOUT,
-            silent=args.short,
-        )
-        if return_code != 0:
-            return return_code
-
-    # Run coverage if needed
-    if args.coverage:
-        shutil.rmtree(COVERAGE_FOLDER, ignore_errors=True)
-        cmd = ["make", "-C", PROJECT_LOCATION, "with_coverage"]
-
-    # Otherwise run the main make command for building and testing
-    else:
-        cmd = ["make", "-C", PROJECT_LOCATION, "bin/c_compiler"]
-
-    return_code = run_subprocess(
-        cmd=cmd,
-        timeout=BUILD_TIMEOUT,
-        silent=args.short,
-    )
-    if return_code != 0:
-        return return_code
-
-    with open(J_UNIT_OUTPUT_FILE, "w") as f:
-        f.write('<?xml version="1.0" encoding="UTF-8"?>\n')
-        f.write('<testsuite name="Integration test">\n')
-
-    drivers = list(Path(args.dir).rglob("*_driver.c"))
-    drivers = sorted(drivers, key=lambda p: (p.parent.name, p.name))
-    log_queue = queue.Queue()
-    results = []
-
-    if sys.stdout.isatty(): # Another TTY fix for Github Actions
-        try:
-            progress_bar = ProgressBar(len(drivers), silent=args.short)
-        except ValueError as e: # Error comes from TTY when running in Github Actions
-            assert not args.short,\
-                "Progress bar cannot be initialised, so program has to be verbose"
-            progress_bar = None
-    else:
-        progress_bar = None
-
-    if args.multithreading:
-        with ThreadPoolExecutor() as executor:
-            futures = [executor.submit(run_test, driver, log_queue)
-                       for driver in drivers]
-
-            for future in as_completed(futures):
-                results.append(future.result())
-                empty_log_queue(log_queue, not args.short, progress_bar)
-
-    else:
-        for driver in drivers:
-            result = run_test(driver, log_queue)
-            results.append(result)
-            empty_log_queue(log_queue, not args.short, progress_bar)
-
-    passing = sum([1 if result == 0 else 0 for result in results])
-    total = len(drivers)
-
-    with open(J_UNIT_OUTPUT_FILE, "a") as f:
-        f.write('</testsuite>\n')
-
-    if not args.short:
-        print(
-            "\n>> Test Summary: " +
-            Fore.GREEN + f"{passing} Passed, " + Fore.RED + f"{total-passing} Failed"
-        )
-
-    # Run coverage if needed
-    if args.coverage:
-        return_code = run_subprocess(
-            cmd=["make", "-C", PROJECT_LOCATION, "coverage"],
-            timeout=BUILD_TIMEOUT,
-            silent=args.short,
-        )
-        if return_code != 0:
-            return return_code
+    if not args.no_clean and not clean():
+        # Clean the repo if required and exit if this fails.
+        exit(1)
 
-    return 0
+    if not make(with_coverage=args.coverage, silent=args.short):
+        exit(1)
```

Comment:
Differentiate from the other ```exit```
```suggestion
        exit(2)
```